### DISCUSSION TITLE
Does support for Custom Resource Configurations in Waypoint?

### DISCUSSION THREAD

**Discussion by nicole-lihui** [istio/istio#51252] [Ideas]
I need to customize the resource configuration for my Waypoint, but I could not find any related configuration options.

From the current implementation of Waypoint, it seems that the template does not support any configuration and is hard-coded. 
https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/files/waypoint.yaml#L185-L191

![image](https://github.com/istio/istio/assets/44200120/d61e1d9a-09c8-4267-af19-b37f21e15076)

This lack of flexibility makes it difficult to adapt Waypoint to specific use cases that require different resource allocations, replica counts, or other pod-level settings.

**Comment by zirain** [comment]
see https://github.com/istio/istio/issues/48000#issuecomment-2136081267

**Comment by danehans** [comment]
This topic was also addressed in the ambient working group [meeting](https://www.google.com/url?q=https://docs.google.com/document/d/1SMlwliEnthgq7r2PjpLl1kCq3t8rAMbgu6r_lDAXJ0w/edit%23heading%3Dh.o8pz6aqnzzgk&sa=D&source=calendar&usd=2&usg=AOvVaw0e9PxEETPmknu7lmQK5olT).
